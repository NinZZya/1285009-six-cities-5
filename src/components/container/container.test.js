import React from 'react';
import renderer from 'react-test-renderer';
import Container from './container';


const CONTAINER_TYPE = `cities__places`;
const CLASS_NAME = `cities-places__container`;

describe(`Should Container render correctly`, () => {
  it(`Should Container render correctly without type and children`, () => {
    const tree = renderer
      .create(
          <Container>
            <div>Div 1</div>
            <div>Div 2</div>
          </Container>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should Container render correctly with type and children`, () => {
    const tree = renderer
      .create(
          <Container type={CONTAINER_TYPE}>
            <div>Div 1</div>
            <div>Div 2</div>
          </Container>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should Container render correctly with type, empty and children`, () => {
    const tree = renderer
      .create(
          <Container empty type={CONTAINER_TYPE}>
            <div>Div 1</div>
            <div>Div 2</div>
          </Container>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should Container render correctly class name and children`, () => {
    const tree = renderer
      .create(
          <Container className={CLASS_NAME}>
            <div>Div 1</div>
            <div>Div 2</div>
          </Container>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
