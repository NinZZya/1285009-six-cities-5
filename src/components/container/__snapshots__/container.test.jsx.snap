// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Container render correctly Should Container render correctly class name and children 1`] = `
<div
  className="container cities-places__container "
>
  <div>
    Div 1
  </div>
  <div>
    Div 2
  </div>
</div>
`;

exports[`Should Container render correctly Should Container render correctly with type and children 1`] = `
<div
  className="container cities__places-container "
>
  <div>
    Div 1
  </div>
  <div>
    Div 2
  </div>
</div>
`;

exports[`Should Container render correctly Should Container render correctly with type, empty and children 1`] = `
<div
  className="container cities__places-container cities__places-container--empty"
>
  <div>
    Div 1
  </div>
  <div>
    Div 2
  </div>
</div>
`;

exports[`Should Container render correctly Should Container render correctly without type and children 1`] = `
<div
  className="container  "
>
  <div>
    Div 1
  </div>
  <div>
    Div 2
  </div>
</div>
`;
